### PR TITLE
enable pydata_sphinx_theme version banner

### LINKED ISSUE #13824: activate pydata-sphinx-theme version banner
Since bokeh 3.2.2 the version banner is broken and a JS error appears in the browser console. See

* https://docs.bokeh.org/en/3.2.2/
* https://docs.bokeh.org/en/3.3.0/
* https://docs.bokeh.org/en/3.4.0/

The error

```
Uncaught ReferenceError: $ is not defined
    <anonymous> https://docs.bokeh.org/en/3.2.2/_static/custom.js:11
```

is caused by

https://github.com/bokeh/bokeh/blob/e380566d129ff16c096a68dec58ba48cfe269939/docs/bokeh/source/_static/custom.js#L10-L26

The pydata-sphinx-theme offers the opportunity to add a version banner. This was mentioned in https://github.com/bokeh/bokeh/issues/13789#issuecomment-2032234165

> Just noting that I believe version warning banners are now built into the theme, so you could try using that and see if that removes > the error
>
> https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/announcements.html#version-warning-banners

The goal of this issue should be to remove the custom solution, silence the JS error and activate the theme banner by pydata-sphinx-theme.

### PR DESCRIPTION
**Pull request by mosc9575**
This PR activates the version banner from pydata-sphinx-theme, removes the old version banner (js and css). The switcher.json in also adapted to silence a warning.

- [ ] issues: fixes #13824

After running `make clean html serve` on my local machine this is the result.

version_banner

For the older versions 

*    https://docs.bokeh.org/en/3.2.2/
*    https://docs.bokeh.org/en/3.3.0/
*    https://docs.bokeh.org/en/3.4.0/

it is suggested to add the code below to all existing html pages to enable the banner.

```

<div style="background-color: rgb(248, 215, 218); color: rgb(114, 28, 36); text-align: center;">
  <div>
    <div>This is documentation for <strong>an old version</strong>.
      <a href="https://docs.bokeh.org/en/latest/index.html" style="background-color: rgb(220, 53, 69); color: rgb(255, 255, 255); margin: 1rem; padding: 0.375rem 0.75rem; border-radius: 4px; display: inline-block; text-align: center;">Switch to stable version</a>
    </div>
  </div>
</div>
```

I don't know how to do this.

### CODE REVIEW

File: docs/bokeh/source/conf.py
Comment (mosc9575) [inline docs/bokeh/source/conf.py]: Here we need some better logic to decide between a local and an online development version.

**Comment by bryevdv** [comment]
> it is suggested to add the code below to all existing html pages to enable the banner.

Unfortunately that would be an enormous ordeal and undertaking. Also potentially costly, as well, as the pages are stored on s3 and we would have to exfiltrate many tens of thousands to update. We would also have to issue very large Cloudfront cache invalidations. I think we will just have to live with old pages being as they are until they are retired from the switcher list. 

Edit: Unless you mean just the top-level `index.html` pages and not anything else. Then sure, we can update a handful of pages manually.

File: docs/bokeh/source/conf.py
Comment (bryevdv) [reply docs/bokeh/source/conf.py]: I'm not too concerned about the `conf.py` contents. Anything quick and easy that works will do

**Comment by mosc9575** [comment]
> Edit: Unless you mean just the top-level `index.html` pages and not anything else. Then sure, we can update a handful of pages manually.

I haven't thought about updating only the start page but I think that would be a good compromise. It is not sufficent to show the version banner on all pages but it is much better than nothing and the start page should have this information.

File: docs/bokeh/source/conf.py
Comment (mosc9575) [reply docs/bokeh/source/conf.py]: How is the process for an offical release? AFAIK the online version for `dev-3.5` would get the wrong `json_url` which is wrong. I just want to make sure that I don't break the development documentation.

Comment (bryevdv) [reply docs/bokeh/source/conf.py]: Ah, I missed that this was the concern. Some ideas:

* usually local docs are / should be build with `BOKEH_DOCS_CDN="local"` set as described [here](https://docs.bokeh.org/en/latest/docs/dev_guide/documentation.html#build-bokeh-s-documentation). I am not sure everyone reliably sets this when doing local dev, though
* actual commands for release scenario are here:
   https://github.com/bokeh/bokeh/blob/d08906b168341db5a4b0fb0b47a1372cd7f0f369/release/build.py#L72

  We could possibly cue off something in the env vars there, or if need be we could add some new make target ("`make release`") to differentiate between local and release build situations

Comment (mosc9575) [reply docs/bokeh/source/conf.py]: If I add `BOKEH_DOCS_RELEASE` to

https://github.com/bokeh/bokeh/blob/d08906b168341db5a4b0fb0b47a1372cd7f0f369/release/build.py#L72

then I can check in the `conf.py` for the existance of `BOKEH_DOCS_RELEASE` and if it exists, it is a release, am I right?

I pushed a commit with this change and I belive the issue is solved. @bryevdv Could you please leave a comment?

Comment (bryevdv) [reply docs/bokeh/source/conf.py]: @mosc9575 I'm just a little reticent to add a third env var in the release call, it's already pretty clunky with two. Thinking a bit more on it, I can't really think of any reason anyone would ever set `BOKEH_DOCS_VERSION` locally, except in the very rare case of a core dev manually rebuilding and redeploying docs, so why don't we just condition on the presence or absence of that? (and In fact if we *are* in that rare case, conditioning off `BOKEH_DOCS_VERSION` will yield the correct behavior more reliably &mdash; it would be trivial to forget some other new var, and that would be incorrect to do).

Otherwise if someone is monkeying round with `BOKEH_DOCS_VERSION` I think we can just assume they know what they are doing and are on their own.

File: docs/bokeh/source/_static/switcher.json
Comment (bryevdv) [inline docs/bokeh/source/_static/switcher.json]: FYI we could also remove this entry. It should have been removed on the last release, we only show the latest point release from each release level, but I forgot to edit it out. But we can try to remember to do it in the next release, too.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
@bokeh/core after this is merged it would be advised to test a dev build release.

**Comment by bryevdv** [comment]
Thanks @mosc9575 FYI I am about to be "out of the office" for a few weeks, maybe make a separate issue regarding updating the old `index.html` pages manually so it doesn't get lost in the shuffle. If you happen to be able to update all the pages locally and store them off somewhere or in a zip file, etc., that I can access, that will definitely speed things up when I get back.

**Comment by mosc9575** [comment]
@bryevdv Below is a zip folder with the adapted index pages. I adapted th current latest version, too. I hope this will work for you.

[bokeh_start_pages.zip](https://github.com/bokeh/bokeh/files/15414759/bokeh_start_pages.zip)

**Pull request merged**
